README.md

```diff
@@ -1,4 +1,45 @@
-## Rusty Merkle Tree Implementation
-**TODO**
-- [ ] Write REAMDE
+# Rusty Merkle Tree Implementation
 
+Simple Merkle tree implementation in Rust.
+
+## Installation
+
+- Clone the repository
+```bash
+git clone https://github.com/LucasUTNFRD/Merkle-Tree.git && cd Merkle-Tree
+```
+
+## Features
+
+- Create Merkle trees from any data that implements `AsRef<[u8]>`
+- Generate Merkle proofs for leaf verification
```

Comment:
Once again, a bit nitpicky; but this paragraph seems a bit off. Specially _"for leaf verification"_.

Maybe expanding on what those verifications mean would be wise. 

What _does_ it verify? It's existence? A set of properties?

Reply:
I made it more simpler now. Let me know if you think i should explain in detail what everything does, i assumed that the one who uses this is someone familiar with Merkle Tree.